presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -419,6 +419,18 @@ private void invokeNoMoreSplitsIfNecessary()
                     false,
                     true));
 
+            if (AcidUtils.isFullAcidTable(table.getParameters())) {
+                /**
+                 * From Hive version >= 3.0, delta/base files will always have file '_orc_acid_version' with value >= '2'.
+                 */
+                Path baseOrDeltaPath = directory.getBaseDirectory() != null ? directory.getBaseDirectory() :
+                        directory.getCurrentDirectories().size() > 0 ? directory.getCurrentDirectories().get(0).getPath() : null;
+
+                if (baseOrDeltaPath != null && AcidUtils.OrcAcidVersion.getAcidVersionFromMetaFile(baseOrDeltaPath, fs) < 2) {
```

Comment:
`getAcidVersionFromMetaFile` call is not wrapped `hdfsEnvironment.doAs` and so i'd expect this to fail in a kerberized env. Yet, the product tests passed. Can you check this gets run with a kerberized HDFS  environment?

Reply:
`getAcidVersionFromMetaFile` takes `fs` (FileSystem) object which gets initialized by using `hdfsAuthentication.doAs() `in `hdfsEnvironment.getFileSystem()`. Therefore, it is not failing.